src/downloadFileTask/downloadFile.ts

```diff
@@ -0,0 +1,31 @@
+import axios from 'axios';
+import fsPromise from 'fs/promises';
+import path from 'path';
+
+const hw3DownloadImageTask = () => {
+
+  const getFileNameFromURL = (path) => {
+    const url = new URL(path);
+    return  url.pathname.split('/').slice(-1).join('');
+  };
+
+  const processImage = async (pathFrom: string, pathToStore: string) => {
+    const fileNameWithExtension = getFileNameFromURL(pathFrom);
+    try {
+      const response = await axios.get(pathFrom, { responseType: 'arraybuffer' });
+      await fsPromise.writeFile(path.join(pathToStore, fileNameWithExtension), response.data);
```

Comment:
what will happen if download folder does not exist 
spoiler
ENOENT: no such file or directory, open '/node_learning/src/downloadFileTask/downloads/image.jpeg'